### ISSUE TITLE
FEATURE REQ: Quarantine compliance proof through the app

### ISSUE THREAD

**Issue by tremblerz** [tripleblindmarket/private-kit#309]
Many city officials require quarantine compliance to be enforced.
One possible solution - Randomly ask for their pictures through the app at anytime in a day (there are few issues around it and hence requires discussion), this picture would get stored along with the timestamp which is already getting logged.
Furthermore, use the successful 14 day quarantine as a verification pass to go back to work.
An additional data science tool which does biometric verification directly would be more helpful as city officials can not check piles of images everytime.

**Comment by E3V3A** [comment]
@tremblerz I don't really get what you're saying. Can you rephrase it somehow to explain how this is supposed to work? (What do you mean with a "picture"?)

---

I think this function would be very useful and an awesome addition, to help health authorities trust people. AFAIK, in China they have an app that basically display a colored "tag" (big round filled circle) in the app that show something like:

1. **`GREEN:`** - The person has either **"recovered"** or doesn't have it. 
1. **`ORANGE:`** - The person has **"unknown"** status or pending test results
1. **`RED:`** - The person is **"infected"** and is under quarantine.

This should be improved upon to separate the green into 2 different results. I suggest:

1. **`BLUE:`** - The person is **"susceptible"** but doesn't have it. (Negative *antigen-test* AND negative *antibody*-test.)
1. **`GREEN:`** - The person has passed an *antibody-based* test. (Independently recovered)

**Comment by StartupSam** [comment]
@E3V3A , related but not quite what @tremblerz is talking about. I think he's mostly referring to a government system to make sure people stay in quarantine and penalize them if they don't.

Main problem I see is an inherent conflict with the privacy-driven ethos of the app. It's certainly something that'd be used though (lots of govs doing this already, as I'm sure you know). Perhaps quarantine-enforcing features could be built on a fork of the project?

Main features I'd envision:
- Picture / geolocation verification of quarantine
- Some way for governments to verify quarantine and penalize those leaving it
- Some kind of badge or verification system that can cryptographically authenticate someone as having stayed in quarantine, and being who they say they are (so they can't just show a screenshot of the success screen)
- Gov / health authority tools for verifying quarantine status (e.g. scanning at checkpoints entering stores or something)

**Comment by E3V3A** [comment]
@StartupSam 
> quarantine-enforcing features ..

I really hate to be someone who say this can't be done. But in this case I'd say both. 

* It can't be practically done in the short time we have, because one can always get another phone, another SIM card, leave the phone at home, force the phone to connect to cell tower far away (even in a different country using virtual SIM's). You can also inject fake GPS coordinates and fudge the GPS location in several different ways. 

* It shouldn't be done, because most research show that when people understand the issues and benefits of honest self reporting, you'll have a much larger compliance (~90%) rather than when it is forced (~70%). [I don't find that reference, atm.]

**Comment by penrods** [comment]
I'm personally opposed to this feature.  This will hinder trust -- that is our key.  The feature is inherently an "I don't trust you" feature.

That should be done in an independent app.

**Comment by andreanuzzo** [comment]
I voiced my concerns in the Slack channel and I will reiterate them here. It's not our role to implement such a feature.

**Comment by E3V3A** [comment]
> That should be done in an independent app.

Just FYI, no app needed. This is already done by "states" security services... 
(I know at least 4 who does this on a continuous and regular basis.)

**Comment by kenpugsley** [comment]
Closing this issue as the consensus is that this will not be done by this app / project.
